router/router.go

```diff
@@ -73,10 +76,13 @@ func New(cfg Config) http.Handler {
 	router.MatcherFunc(isKnownBabbageEndpointMatcher).Handler(cfg.BabbageHandler)
 
 	// all other requests go through the allRoutesMiddleware to check the page type first
-	allRoutesMiddleware := allRoutes.Handler(map[string]http.Handler{
+	handlers := map[string]http.Handler{
 		"dataset_landing_page": cfg.DatasetHandler,
-	}, cfg.ZebedeeClient, cfg.ContentTypeByteLimit)
-
+	}
+	if cfg.DatasetEnabled {
```

Comment:
Consider adding a router test for this, to ensure we route as expected when the flag is enabled / disabled.

Reply:
I think you may need to walk me through on how to do that.

Reply:
Added.